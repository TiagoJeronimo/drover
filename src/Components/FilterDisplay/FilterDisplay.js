import React, { Component } from 'react';
import './FilterDisplay.style.scss';
import FilterDropdown from '../FilterDropdown/FilterDropdown';
import FilterInput from '../FilterInput/FilterInput';

const filtersDropdownOptions = [
  {
    key: 'vehicle_make',
    text: 'Car Make',
    aggregationKey: 'vehicle_make'
  },
  {
    key: 'transmission',
    text: 'Gearbox',
    aggregationKey: 'transmission'
  },
  {
    key: 'fuel',
    text: 'Fuel Type',
    aggregationKey: 'fuel'
  },
  {
    key: 'body_type',
    text: 'Body Type',
    aggregationKey: 'body_information'
  }
];

class FilterDisplay extends Component {
  state = {
    location: ''
  };

  formatOptions = (options, showQuantity = true) => {
    const optionArray = [{ text: 'Any', value: 'any' }];
    let text = '';
    Object.keys(options).forEach(key => {
      text = `${key} ${showQuantity && `(${options[key]})`}`;
      optionArray.push({ text: text, value: key });
    });
    return optionArray;
  };

  handleInputChange = (name, value) => {
    this.setState({
      [name]: value
    });
  };

  render() {
    const { aggregations, handleFilterChange, selectedFilters } = this.props;

    return (
      <div className="filterDisplay-container">
        {filtersDropdownOptions.map(filter => (
          <FilterDropdown
            key={filter.key + filter.text}
            filterName={filter.key}
            labelText={filter.text}
            dropdownOptions={this.formatOptions(aggregations[filter.aggregationKey])}
            handleFilterChange={(e, { name, value }) => handleFilterChange(name, value)}
            selectedFilter={!selectedFilters[filter.key] ? 'Any' : ''}
          />
        ))}
        <FilterInput
          labelText="Location"
          inputFieldName="location"
          onClickInputField={() => handleFilterChange('location', this.state.location)}
          handleInputChange={this.handleInputChange}
        />
      </div>
    );
  }
}

export default FilterDisplay;
