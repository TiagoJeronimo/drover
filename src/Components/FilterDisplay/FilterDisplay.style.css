.filterDisplayContainer {
  max-width: 25%;
  flex: 1; }
