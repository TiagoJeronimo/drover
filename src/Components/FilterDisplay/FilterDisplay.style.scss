.filterDisplay-container {
  margin: 0 15px;
  max-width: 25%;
  flex: 1;
}
