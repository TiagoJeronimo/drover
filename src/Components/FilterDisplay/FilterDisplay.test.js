import React from 'react';
import { mount } from 'enzyme';

import FilterDisplay from './FilterDisplay';

it('FilterDisplay with normal props', () => {
  const props = {
    aggregations: {
      vehicle_make: {
        Peugeot: 60
      },
      vehicle_model_group: {
        '108': 20
      },
      sub_type: {
        '': 176
      },
      fuel: {
        petrol: 100
      },
      transmission: {
        manual: 110
      },
      year: {
        '2015': 2
      },
      tags: {
        compact: 24
      },
      body_information: {
        hatchback: 86
      },
      city_jurisdiction: {
        london: 66
      }
    },
    //filtersDropdownOptions: [{text: 'Any', value: 'any'}, {text: 'Car', value: 'car'}],
    handleFilterChange: jest.fn(),
    selectedFilters: { vehicle_make: 'car' }
  };
  const component = <FilterDisplay {...props} />;
  const html = mount(component).html();
  expect(html).toMatchSnapshot();
});
