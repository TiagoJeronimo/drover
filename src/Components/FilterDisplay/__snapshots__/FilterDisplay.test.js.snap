// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDisplay with normal props 1`] = `"<div class=\\"filterDisplay-container\\"><div class=\\"filterDropdown-container\\"><div class=\\"filterDropdown-label\\">Car Make</div><div name=\\"vehicle_make\\" role=\\"listbox\\" aria-expanded=\\"false\\" class=\\"ui fluid floating selection dropdown\\" tabindex=\\"0\\"><div class=\\"default text\\" role=\\"alert\\" aria-live=\\"polite\\">Any</div><i aria-hidden=\\"true\\" class=\\"dropdown icon\\"></i><div class=\\"menu transition\\"><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"true\\" class=\\"selected item\\"><span class=\\"text\\">Any</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">Peugeot (60)</span></div></div></div></div><div class=\\"filterDropdown-container\\"><div class=\\"filterDropdown-label\\">Gearbox</div><div name=\\"transmission\\" role=\\"listbox\\" aria-expanded=\\"false\\" class=\\"ui fluid floating selection dropdown\\" tabindex=\\"0\\"><div class=\\"default text\\" role=\\"alert\\" aria-live=\\"polite\\">Any</div><i aria-hidden=\\"true\\" class=\\"dropdown icon\\"></i><div class=\\"menu transition\\"><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"true\\" class=\\"selected item\\"><span class=\\"text\\">Any</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">manual (110)</span></div></div></div></div><div class=\\"filterDropdown-container\\"><div class=\\"filterDropdown-label\\">Fuel Type</div><div name=\\"fuel\\" role=\\"listbox\\" aria-expanded=\\"false\\" class=\\"ui fluid floating selection dropdown\\" tabindex=\\"0\\"><div class=\\"default text\\" role=\\"alert\\" aria-live=\\"polite\\">Any</div><i aria-hidden=\\"true\\" class=\\"dropdown icon\\"></i><div class=\\"menu transition\\"><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"true\\" class=\\"selected item\\"><span class=\\"text\\">Any</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">petrol (100)</span></div></div></div></div><div class=\\"filterDropdown-container\\"><div class=\\"filterDropdown-label\\">Body Type</div><div name=\\"body_type\\" role=\\"listbox\\" aria-expanded=\\"false\\" class=\\"ui fluid floating selection dropdown\\" tabindex=\\"0\\"><div class=\\"default text\\" role=\\"alert\\" aria-live=\\"polite\\">Any</div><i aria-hidden=\\"true\\" class=\\"dropdown icon\\"></i><div class=\\"menu transition\\"><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"true\\" class=\\"selected item\\"><span class=\\"text\\">Any</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">hatchback (86)</span></div></div></div></div><div class=\\"filterInput-container\\"><div class=\\"filterInput-label\\">Location</div><div class=\\"ui action input\\"><input placeholder=\\"London, Uk\\" name=\\"location\\" type=\\"text\\"><button class=\\"ui icon button\\"><i aria-hidden=\\"true\\" class=\\"search icon\\"></i></button></div></div></div>"`;
